src/types.ts

```diff
@@ -24,13 +24,23 @@ export type Member<T> = T extends (infer arrayMember)[] ? arrayMember : T[keyof
 //** Get the keys of an object where the values match a given type */
 export type TypedKeys<Obj, T> = { [K in keyof Obj]-?: Obj[K] extends T ? K : never }[keyof Obj];
 
-/** Status object for functions that return a boolean indicating success/failure
- * and an optional message */
+/**
+ * Status object for functions that return a boolean indicating success/failure
+ * and an optional message
+ */
 export interface IReturnStatus {
   res: boolean;
   msg?: string;
 }
 
+type AvailabilitySuccess<T extends object> = { available: true } & T;
+type AvailabilityFailure = { available?: undefined; error: string };
+export type Availability<T extends object = object> = AvailabilitySuccess<T> | AvailabilityFailure;
+
+type AttemptSuccess<T extends object> = { success: true; message?: string } & T;
```

Comment:
These two are the same type, just with different labels. I think we should try to standardize on one, and "gently" deprecate the other. (I.e., leave it in Bladeburner.)